src/zenml/models/component_models.py

```diff
@@ -0,0 +1,92 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Models representing stack components."""
+
+from typing import Any, Dict
+
+from pydantic import BaseModel, Field, validator
+
+from zenml.enums import StackComponentType
+from zenml.logger import get_logger
+from zenml.models.base_models import (
+    ShareableRequestModel,
+    ShareableResponseModel,
+    update,
+)
+from zenml.models.constants import MODEL_NAME_FIELD_MAX_LENGTH
+from zenml.utils import secret_utils
+
+logger = get_logger(__name__)
+
+# TODO: Add example schemas and analytics fields
```

Comment:
I just wanted to comment on this exact thing, but then I saw the TODO here. At least the analytics fields are a must-have for this PR to be mergeable.